security/certManager/certManager.go

```diff
@@ -0,0 +1,195 @@
+package certManager
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"strings"
+	"sync"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/plgd-dev/kit/security"
+)
+
+// CertManager holds certificates from filesystem watched for changes
+type FileCertManager struct {
+	mutex                   sync.Mutex
+	config                  Config
+	tlsKey                  []byte
+	tlsCert                 []byte
+	tlsKeyPair              tls.Certificate
+	certificateAuthorities  []*x509.Certificate
+	watcher                 *fsnotify.Watcher
```

Comment:
watcher per file path (ca-file, crt-file, key-file) ofcourse you can merge it when directory is same.